Update OCP-4-13-release-notes for Developer Perspective: Known Issue (#58236)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * Currently, when adding a Git Repository, the created repository resource is invalid if you configure it with a GitLab and Bitbucket `pipeline-as-a-code` repository.

Reviewer: @shreyasiddhartha Requires rewrite. Are the steps sequential here?

Typically, workarounds are written as instructions and in the present tense. Please see the example in the template.

Reviewer: @shreyasiddhartha Also in the OCPBUGS-7036 description I see this sentence: The spec.git_provider.url should be https://api.bitbucket.org/2.0, or **can be removed completely. Both work fine.**

But the RN text Workaround in your PR and the solution description in the engineering PR #12593 says it is mandatory. I do not find this information in the OCP JIRA. This is confusing. Please consult with the SME and clarify.

Reviewer: @shreyasiddhartha Is the `spec.git_provider.user` field and the **Bitbucket username** field different from each other? Both are mandatory? Please consult the SMEs and confirm.

It seems that they are the same for me, but not sure.

Author: @Srivaralakshmi I have made some changes after consulting with the SME. PTAL and let me know if this looks good.

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * When adding a Git Repository, the created repository resource was invalid if you configured it with a GitLab and Bitbucket `pipeline-as-a-code` repository. Now, the Git provider URL is removed for GitLab and Bitbucket providers. A mandatory *Bitbucket username* field is added if the provider is Bitbucket. You also need to select either *Git access token* or *Git access token secret* to continue adding a Git Repository. (link:https://issues.redhat.com/browse/OCPBUGS-7036[*OCPBUGS-7036*])

Reviewer: @shreyasiddhartha I see that the issue here is that the created repository resource was invalid. Are the remaining sentences the workaround for this issue? Or they are also part of the Known issue?

Also, I do not see any mention of this: spec.git_provider.user details for BitBucket provider. Any particular reason why this is not mentioned?

Reviewer: @shreyasiddhartha Also please note that this is the template RH SSG suggests for writing the RN Texts: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes-doc-texts

Please adhere to these guidelines and standards when documenting RN Texts for KIs or KIs with workarounds. Thanks!

Author: @Srivaralakshmi I have made a few changes as per your suggestions. PTAL. Thanks!

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * Currently, when adding a Git Repository, the created repository resource is invalid if you configure it with a GitLab and Bitbucket `pipeline-as-a-code` repository. As a result, the Git provider URL `spec.git_provider.url` is removed for GitLab and Bitbucket providers. With this workaround, you use the mandatory *Bitbucket username* field for Bitbucket and select either *Git access token* or *Git access token secret* to continue adding a Git Repository. (link:https://issues.redhat.com/browse/OCPBUGS-7036[*OCPBUGS-7036*])

Reviewer: ```suggestion
* Adding a Git Repository and configuring it with a GitLab and Bitbucket `pipeline-as-a-code` repository creates an invalid repository resource. As a result, the `spec.git_provider.url` Git provider URL is removed for GitLab and Bitbucket providers.

Workaround: Add the mandatory `spec.git_provider.user` field for Bitbucket. In addition, select either *Git access token* or *Git access token secret* to continue adding a Git Repository. (link:https://issues.redhat.com/browse/OCPBUGS-7036[*OCPBUGS-7036*])
```
